src/containers/FilterBarContainer.tsx

```diff
@@ -0,0 +1,50 @@
+import { connect } from "react-redux";
+import * as React from "react";
+import { Store, asyncActions, facetsActions, searchParameterActions } from "azsearchstore";
+import * as redux from "redux";
+import FilterBar from "../components/FilterBar";
+
+function getReturnType<RT>(expression: (...params: any[]) => RT): RT {
+    return {} as RT;
+}
+
+export interface OwnProps {
+}
+
+const mapDispatchToProps = (dispatch: redux.Dispatch<any>, ownProps: OwnProps) => {
+  return {
+    onClear: () => {
+      dispatch(facetsActions.clearFacetsSelections);
+      dispatch(searchParameterActions.setPage(1));
+      dispatch(asyncActions.fetchSearchResults);
+    }
+  };
+};
+
+function mapStateToProps(state: Store.SearchState, ownProps: OwnProps) {
+  return {
+    hasSelectedFacets: checkForAppliedFacets(state.facets.facets)
+  };
+}
+
+function checkForAppliedFacets(facets: { [key: string]: Store.Facet }) {
+  for (let facetKey in facets) {
+    let checkboxFacet = facets[facetKey] as Store.CheckboxFacet;
```

Comment:
Wow totally missed that but yes, poor assumption on my part. Would be best to remove the logic entirely and have the store return if a query is filtered or not (or use the filter string itself) to determine if the element should be shown? Otherwise I can do the switch like you have in the store